### PR TITLE
use public API to sync users and groups

### PR DESCRIPTION
**Pull request by jfshenuw**
There were a few changes to data structure (user.id instead user.userName, for example) and url path from the service API to public API.

The changes were tested by recreating my account and by sync'ing test users.

### CODE REVIEW

File: src/main/node/services/acs.ts [hunk 1]
```diff
@@ -75,35 +75,55 @@ export class ACS {
...
+            retv = JSON.parse(body);
+            if ( retv && retv.status && retv.status.code && retv.status.code==HTTP_STATUS_CODE_FOR_NOT_FOUND ) {
+                retv = null;
+            }
+        })
+        .catch((err) => {
+            if ( err.statusCode && err.statusCode == HTTP_STATUS_CODE_FOR_NOT_FOUND) {
+                retv = null;
+            } else {
+                this.logError(err, 'ERROR - getUserGroups returned error for user ' + username)
```
Comment (agagne) [inline src/main/node/services/acs.ts]: Should we include the status code?

File: src/main/node/services/acs.ts [hunk 2]
```diff
@@ -42,17 +42,17 @@ export class ACS {
             if ( err.statusCode == 409 ) {
                 // user already exists, noop
             } else {
-                this.logError(err, 'ERROR - createUser returned error for user ' + user.userName);
+                this.logError(err, 'ERROR - createUser returned error for user ' + user.id);
```
Comment (agagne) [inline src/main/node/services/acs.ts]: should we add the status code?

File: src/main/node/services/acs.ts (hunk 1 above)
Comment (jfshenuw) [reply src/main/node/services/acs.ts]: I think the status code is included as part of "err"

File: src/main/node/services/acs.ts (hunk 2 above)
Comment (jfshenuw) [reply src/main/node/services/acs.ts]: the status code is part of "err"

File: src/main/node/services/acs.ts [hunk 3]
```diff
@@ -75,35 +75,55 @@ export class ACS {
...
             auth: this.auth
         }
         await this.request.get(options, (err, response, body) => {
-            retv = JSON.parse(body).data;
+            retv = JSON.parse(body);
         })
         .catch((err) => {
-                this.logError(err, 'ERROR 2 - getMembers returned error for group ' + groupId)
-                throw(err);
+            this.logError(err, 'ERROR 2 - getMembers returned error for group ' + groupId)
```
[the remark this answers is not in the document]
Comment (jfshenuw) [reply src/main/node/services/acs.ts]: Forgot to remove "2" after testing. It is removed now.

**Pull request merged**